core_v2/sources/v2_config.move

```diff
@@ -0,0 +1,476 @@
+/*
+Provides a singleton wrapper around PropertyMap to allow for easy and dynamic configurability of contract options.
+This includes things like the maximum number of years that a name can be registered for, etc.
+
+Anyone can read, but only admins can write, as all write methods are gated via permissions checks
+*/
+
+module aptos_names_v2::v2_config {
+    friend aptos_names_v2::v2_domains;
+
+    use aptos_framework::account;
+    use aptos_framework::aptos_account;
+    use aptos_std::ed25519::{Self, UnvalidatedPublicKey};
+    use aptos_names_v2::v2_string_validator;
+    use aptos_token::property_map::{Self, PropertyMap};
+    use std::error;
+    use std::signer;
+    use std::string::{Self, String};
+    use std::vector;
+
+    const CONFIG_KEY_ENABLED: vector<u8> = b"enabled";
+    const CONFIG_KEY_ADMIN_ADDRESS: vector<u8> = b"admin_address";
+    const CONFIG_KEY_FUND_DESTINATION_ADDRESS: vector<u8> = b"fund_destination_address";
+    const CONFIG_KEY_MAX_NUMBER_OF_SECONDS_REGISTERED: vector<u8> = b"max_number_of_seconds_registered";
+    const CONFIG_KEY_MAX_DOMAIN_LENGTH: vector<u8> = b"max_domain_length";
+    const CONFIG_KEY_MIN_DOMAIN_LENGTH: vector<u8> = b"min_domain_length";
+    const CONFIG_KEY_TOKENDATA_DESCRIPTION: vector<u8> = b"tokendata_description";
+    const CONFIG_KEY_TOKENDATA_URL_PREFIX: vector<u8> = b"tokendata_url_prefix";
+    const CONFIG_KEY_DOMAIN_PRICE_PREFIX: vector<u8> = b"domain_price_";
+    const CONFIG_KEY_SUBDOMAIN_PRICE: vector<u8> = b"subdomain_price";
+    /// The number of seconds after a name expires that it can be re-registered
+    const CONFIG_KEY_REREGISTRATION_GRACE_SEC: vector<u8> = b"reregistration_grace_sec";
+
+    const DOMAIN_TYPE: vector<u8> = b"domain";
+    const SUBDOMAIN_TYPE: vector<u8> = b"subdomain";
+
+    const DOMAIN_COLLECTION_NAME: vector<u8> = b"Aptos Domain Names V2";
+    const SUBDOMAIN_COLLECTION_NAME: vector<u8> = b"Aptos Subdomain Names V2";
+
+    /// Raised if the signer is not authorized to perform an action
+    const ENOT_AUTHORIZED: u64 = 1;
+    /// Raised if there is an invalid value for a configuration
+    const EINVALID_VALUE: u64 = 2;
+
+    const SECONDS_PER_YEAR: u64 = 60 * 60 * 24 * 365;
+
+    struct Config has key, store {
+        config: PropertyMap,
+    }
+
+    public(friend) fun initialize_config(
+        framework: &signer,
```

Comment:
nit: This is a bit misleading as framework signer often means 0x1 signer but this is not it